bbc/simorgh
# WSTEAM1-A11y-Bug-Fix: Remove extra space from live label

Resolves JIRA [WSTEAM1-846](https://jira.dev.bbc.co.uk/browse/WSTEAM1-846)

Overall changes
======
Refactor Live Label logic so that extra space is not added to the end of Live Label if no Offscreen Text is provided.

It's not clear whether this was causing an A11y bugs, however it is generally good practice to remove redundant* spaces.

*(To confirm with Team that this space is indeed redundant)

Code changes
======

- Update logic src/app/components/LiveLabel/Text/index.tsx so that space appends optional offscreen text.
- Fix unit tests by moving space from Live Label values to Offscreen Text values
- Fix unit tests by updating 'getByText' value
- Update snapshots

Testing
======

1. Open the below links to the PR permalinks and latest
2. Check there is no visual change
3. Disable CSS and confirm that the space before the comma has been removed when there is no offscreen text. Check there is no change when Offscreen text is provided.

- This PR: [Live Label (with custom offscreen text)](https://a11y-bug-fix-extra-space-live-label--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=new-components-live-label--with-custom-offscreen-text&viewMode=story) vs [Latest](https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=new-components-live-label--with-custom-offscreen-text&viewMode=story)
- This PR: [Live Label (with children)](https://a11y-bug-fix-extra-space-live-label--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=new-components-live-label--with-children&viewMode=story) vs [Latest](https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=new-components-live-label--with-children&viewMode=story)
- This PR: [Live Label with custom offscreen text and children](https://a11y-bug-fix-extra-space-live-label--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&knob-Select%20a%20service=russian&id=new-components-live-label--with-children&viewMode=story) vs [Latest](https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&knob-Select%20a%20service=russian&id=new-components-live-label--with-children&viewMode=story)
- This PR: [Live Page header with label (russian, with offscreen text)](https://a11y-bug-fix-extra-space-live-label--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&knob-Select%20a%20service=russian&id=new-components-livepageheader--title-only-with-live-label&viewMode=story) vs [Latest](https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&knob-Select%20a%20service=russian&id=new-components-livepageheader--title-only-with-live-label&viewMode=story)
- This PR: [Live Page header with label (pidgin, no offscreen text)](https://a11y-bug-fix-extra-space-live-label--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&knob-Select%20a%20service=pidgin&id=new-components-livepageheader--title-only-with-live-label&viewMode=story) vs [Latest](https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&knob-Select%20a%20service=pidgin&id=new-components-livepageheader--title-only-with-live-label&viewMode=story)
- This PR: [Live label (localised)](https://a11y-bug-fix-extra-space-live-label--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=new-components-live-label--localised&viewMode=story) vs [Latest](https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=new-components-live-label--localised&viewMode=story)
- This PR: [Live TV Bulletin](https://a11y-bug-fix-extra-space-live-label--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&knob-Select%20a%20service=pidgin&id=containers-bulletin--live-tv-bulletin&viewMode=story) vs [Latest](https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&knob-Select%20a%20service=pidgin&id=containers-bulletin--live-tv-bulletin&viewMode=story)

To confirm with Team there are no other instances where this space might be required.

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
